packages/lesswrong/server/callbacks/postCallbacks.ts

```diff
@@ -1,754 +1,259 @@
 }
 
-if (HAS_EMBEDDINGS_FOR_RECOMMENDATIONS) {
-  const updateEmbeddings = async (newPost: DbPost, oldPost?: DbPost) => {
-    const hasChanged = !oldPost || oldPost.contents_latest !== newPost.contents_latest;
-    if (hasChanged &&
-      !newPost.draft &&
-      !newPost.deletedDraft &&
-      newPost.status === postStatuses.STATUS_APPROVED &&
-      !isAnyTest
-    ) {
-      try {
-        await updatePostEmbeddings(newPost._id);
-      } catch (e) {
-        // We never want to prevent a post from being created/edited just
-        // because we fail to create embeddings, but we do want to log it
-        captureException(e);
-        // eslint-disable-next-line
-        console.error("Failed to create embeddings:", e);
-      }
-    }
-  }
-  getCollectionHooks("Posts").newAsync.add(
-    async (post: DbPost) => await updateEmbeddings(post),
-  );
-  getCollectionHooks("Posts").updateAsync.add(
-    async ({document, oldDocument}) => await updateEmbeddings(
-      document,
-      oldDocument,
-    ),
-  );
-}
+async function postCreateBefore(doc: DbInsertion<DbPost>, props: CreateCallbackProperties<'Posts'>): Promise<DbInsertion<DbPost>> {  
+  let mutablePost = doc;
 
-async function checkRecentRepost(post: DbPost): Promise<DbPost> {
-  if (!post.draft) {
-    const oneHourAgo = new Date(Date.now() - (60 * 60 * 1000));
-    const existing = await Posts.findOne({
-      _id: {$ne: post._id},
-      title: post.title,
-      userId: post.userId,
-      draft: {$ne: true},
-      deletedDraft: {$ne: true},
-      createdAt: {$gt: oneHourAgo},
-    });
-    if (existing) {
-      throw new Error(`You recently published another post titled "${post.title}"`);
-    }
-  }
-  return post;
-}
+  // mutablePost = await slugCreateBeforeCallbackFunction(mutablePost, props);
+  mutablePost = addReferrerToPost(mutablePost, props) ?? mutablePost;
+  
+  // mutablePost = await editorSerializationBeforeCreate(mutablePost, props, postContentsEditableCallbackOptions);
+  // mutablePost = await editorSerializationBeforeCreate(mutablePost, props, postModerationGuidelinesEditableCallbackOptions);
+  // mutablePost = await editorSerializationBeforeCreate(mutablePost, props, postCustomHighlightEditableCallbackOptions);
 
-getCollectionHooks("Posts").newSync.add(checkRecentRepost);
-getCollectionHooks("Posts").updateBefore.add(async (data, { newDocument }) => {
-  await checkRecentRepost(newDocument);
-  return data;
-});
+  return mutablePost;
+}
 
-getCollectionHooks("Posts").createValidate.add(function DebateMustHaveCoauthor(validationErrors, { document }) {
-  if (document.debate && !document.coauthorStatuses?.length) {
-    throw new Error('Dialogue must have at least one co-author!');
+async function postNewSync(post: DbPost, currentUser: DbUser | null, context: ResolverContext): Promise<DbPost> {
+  // TODO: add forum-gated EA forum callbacks
+  if (isEAForum) {
+    // TODO: were the errors thrown by these previously being swallowed?
```

Comment:
I'm not sure if the errors thrown by these asserts were actually getting swallowed and otherwise not interrupting the post creation, before this.